sync-s3.sh

```diff
@@ -92,3 +100,17 @@ done
 echo "Syncing app-info to $S3_BUCKET"
 aws s3 sync "${AWS_OPTS[@]}" --region "$S3_REGION" \
     "$SRCDIR/s3/app-info" "s3://$S3_BUCKET/app-info"
+
+# Invalidate the CloudFront distribution so it fetches new versions of
+# any files.
+#
+# FIXME: This is very wasteful as it's likely most of the files haven't
```

Comment:
Wasteful in computation. I'm pretty sure that transfers from S3 to CloudFront are free, so it's really just slowness for clients as they'll have to wait for CloudFront to refetch assets from S3 that would be fine to be cached. Since (I believe) this CloudFront distribution is currently using the default of 1 day TTLs and we're unlikely to run this more than once a day, it probably makes no difference right now. However, we could change the default TTL to be much longer like a week if we're reliably invalidating on updates. Almost all the data here is essentially static.

Reply:
There is also a minor cost for excessive invalidation. From [the pricing page](https://aws.amazon.com/cloudfront/pricing/):

> Invalidation requests
>
> No additional charge for the first 1,000 paths requested for invalidation each month. Thereafter, $0.005 per path requested for invalidation. 
>
> Note: A path listed in your invalidation request represents the URL (or multiple URLs if the path contains a wildcard character) of the object(s) you want to invalidate from CloudFront cache. For more information about invalidation, see [Invalidating Objects](http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/Invalidation.html) in the Amazon CloudFront Developer Guide.

Since there are currently nearly 3000 objects in the S3 bucket, I guess we'd quickly overrun the free allotment. However, I think with the wildcard it depends on what paths are actually in the CloudFront cache. If no one has requested some ancient screenshot, then I can't see how CloudFront would know that it needs to be invalidated.

Reply:
So that would mean $10 each time we push to this repo? 😬

Reply:
Hmm, let me check on my understanding of the wildcard handling. Hopefully there should be some billing data from when I did this yesterday as a test.

Reply:
I think it must be per actual specified path, so `/*` counts as 1. Yesterday our cloudfront costs were less than $0.00. In fact, our entire year to date cost for cloudfront is $10.74. That was all in the helpcenter on Jan 25 & 26. We have yet to spend anything in February.